subsys/net/l2/ethernet/arp.c

```diff
@@ -259,6 +259,12 @@ static inline struct net_pkt *arp_prepare(struct net_if *iface,
 						sizeof(struct net_arp_hdr),
 						AF_UNSPEC, 0, NET_BUF_TIMEOUT);
 		if (!pkt) {
+			if (entry) {
+				/* Add the arp entry back to arp_free_entries, to avoid
+				 * the arp entry is leak due to direct return here.
+				 */
+				sys_slist_prepend(&arp_free_entries, &entry->node);
+			}
```

Comment:
I would prefer that this is done in the same function that allocated the entry so in `net_arp_prepare()`.

Reply:
thanks, will update it